controllers/component_controller.go

```diff
@@ -347,6 +349,56 @@ func (r *ComponentReconciler) runBuild(ctx context.Context, component *appstudio
 	}
 	log.Info(fmt.Sprintf("PV is now present : %v", workspaceStorage.Name))
 
+	// Fetch the SPI Secret if available
+	spiTokenBindings := &spiapi.SPIAccessTokenBindingList{}
+	err = r.List(ctx, spiTokenBindings, &client.ListOptions{Namespace: component.Namespace})
+	if err != nil {
+		log.Error(err, fmt.Sprintf("Unable to list SPI Bindings in namespace %s", component.Namespace))
+	}
+
+	var sourceSecretName string
+	for _, spiTokenBinding := range spiTokenBindings.Items {
+		if spiTokenBinding.Spec.RepoUrl == component.Spec.Source.GitSource.URL {
+			sourceSecretName = spiTokenBinding.Status.SyncedObjectRef.Name
+			log.Info(fmt.Sprintf("Found SPITokenBinding for %s : %s", component.Spec.Source.GitSource.URL, sourceSecretName))
+
+			// Make the Secret ready for consumption by Tekton.
+			gitSecret := corev1.Secret{}
+			err = r.Get(ctx, types.NamespacedName{Name: sourceSecretName, Namespace: component.Namespace}, &gitSecret)
+			if err != nil {
+				log.Error(err, fmt.Sprintf("Secret %s is missing", sourceSecretName))
+				return err
+			} else {
+				if gitSecret.Annotations == nil {
+					gitSecret.Annotations = map[string]string{}
+				}
+				// TODO: Support generic Git providers.
```

Comment:
If I am not mistaken, we can just use protocol + hostname from the repoUrl

Reply:
Yes 👍